File: lmFabula (copy)/Section.java
import java.util.ArrayList;

public class Section
{
	ArrayList<Section> mNestedSections = new ArrayList<Section>();
	ArrayList<Scene>   mScenes         = new ArrayList<Scene>();
	String mName = new String();

	Section()
	{

	}

	void addSection(Section section)
	{
		mNestedSections.add(section);
	}

	void addScene(Scene scene)
	{
		mScenes.add(scene);
	}

	Section findSection(String name)
	{
		for (int i = 0; i < mNestedSections.size(); ++i)
			if (mNestedSections.get(i).mName.equals(name))
				return mNestedSections.get(i);
		return null;
	}


	Scene findScene(String name)
	{
		for (int i = 0; i < mScenes.size(); ++i)
			if (mScenes.get(i).getName().equals(name))
				return mScenes.get(i);
		return null;
	}

	String getName()
	{
		return mName;
	}

}

File: lcFabula/src/Parser/ParseNodes/section.cpp
#include "section.h"
#include "scene.h"

namespace fabula
{
    namespace parsing
    {
        namespace node
        {
            Section::Section() {}

            Section::~Section()
            {
                for (Scene* s : mScenes)
                    delete s;
                for (Section* s : mSubsections)
                    delete s;
            }

            void Section::name(const std::string& name)
            {
                mName = name;
            }

            std::string Section::name() const
            {
                return mName;
            }

            void Section::add(Section* s)
            {
                mSubsections.push_back(s);
            }

            void Section::add(Scene* s)
            {
                mScenes.push_back(s);
            }
        }
    }
}